main-app/app/scripts/bingoTicket/bingoModel.js

```diff
@@ -6,32 +6,47 @@
             me.tickets = [];
             me.bingoStrip = [];
 
-            var clearOldTickets = function () {
-                me.tickets = [];
-                me.bingoStrip = [];
-            };
-
             me.getTicket = function (ticketFromServer) {
                 clearOldTickets();
                 generateTickets(ticketFromServer);
-                for (var i = 0; i < me.tickets.length; i++) {
+                for(var i = 0; i < me.tickets.length; i++){
                     createBingoTicketGrid();
                     splitingNumbersUp(me.tickets[i]);
                 }
             };
 
+            var clearOldTickets = function () {
+                me.tickets = [];
+                me.bingoStrip = [];
+            };
+
+            var BingoCell = function () {
+                this.bingoNumber = null;
+                this.cssClass = "unmarked";
+                this.counted = false;
+            };
+
+            var TicketInfo = function () {
```

Comment:
There has been a trick missed here:
Just as you have a  `BingoCell`, a `TicketLine` would be useful - it contains an array of `BingoCell` and can keep track of how many cells are to go via their incipient `isToGo()` functions